File: test/conv-test-jv/src/test/java/example/EnumMemberRenameTest.java
package example;

// Regression test for the `EnumVariant : was[OldName]` rename conversion
// (see test/conv-test/pkg02.baboon `enum EnumMemberRename` and the matching
// Rust/TypeScript regression tests in test/conv-test-{rs,ts}/tests/).
// Invokes the generated rename conversion against:
//   1. a renamed source variant (OldValue -> NewValue);
//   2. a non-renamed variant (KeepValue) — passes through the fallback
//      arm of the switch in the generated mapper.

import convtest.testpkg.BaboonConversions;
import convtest.testpkg.Convert__EnumMemberRename__From__1_0_0;
import convtest.testpkg.EnumMemberRename;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class EnumMemberRenameTest {

    private final BaboonConversions convs = new BaboonConversions();

    @Test
    public void renamedVariantMapsToNewName() {
        EnumMemberRename mapped = Convert__EnumMemberRename__From__1_0_0.INSTANCE.doConvert(
            (Object) null,
            convs,
            convtest.testpkg.v1_0_0.EnumMemberRename.OldValue
        );
        assertEquals(EnumMemberRename.NewValue, mapped);
    }

    @Test
    public void nonRenamedVariantPassesThrough() {
        EnumMemberRename mapped = Convert__EnumMemberRename__From__1_0_0.INSTANCE.doConvert(
            (Object) null,
            convs,
            convtest.testpkg.v1_0_0.EnumMemberRename.KeepValue
        );
        assertEquals(EnumMemberRename.KeepValue, mapped);
    }
}

File: test/conv-test-dt/test/enum_member_rename_test.dart
// Regression test for the `EnumVariant : was[OldName]` rename conversion
// (see test/conv-test/pkg02.baboon `enum EnumMemberRename` and the matching
// Rust/TypeScript regression tests in test/conv-test-{rs,ts}/tests/).
// Invokes the generated rename conversion against:
//   1. a renamed source variant (OldValue -> NewValue);
//   2. a non-renamed variant (KeepValue) — passes through the fallback
//      arm of the generated mapper.

import 'package:conv_test_dt/generated/convtest/testpkg/baboon_conversions.dart';
import 'package:conv_test_dt/generated/convtest/testpkg/enum_member_rename.dart';
import 'package:conv_test_dt/generated/convtest/testpkg/from-1.0.0-enum_member_rename.dart';
import 'package:conv_test_dt/generated/convtest/testpkg/1.0.0/enum_member_rename.dart'
    as v1_0_0_enum_member_rename;
import 'package:test/test.dart';

void main() {
  final convs = BaboonConversions();

  test('renamed variant OldValue maps to NewValue', () {
    final mapped = Convert__EnumMemberRename__From__1_0_0.instance.doConvert(
      null,
      convs,
      v1_0_0_enum_member_rename.EnumMemberRename.OldValue,
    );
    expect(mapped, equals(EnumMemberRename.NewValue));
  });

  test('non-renamed variant KeepValue passes through', () {
    final mapped = Convert__EnumMemberRename__From__1_0_0.instance.doConvert(
      null,
      convs,
      v1_0_0_enum_member_rename.EnumMemberRename.KeepValue,
    );
    expect(mapped, equals(EnumMemberRename.KeepValue));
  });
}
